src/main/java/frc/trigon/robot/subsystems/turret/TurretIO.java

```diff
@@ -0,0 +1,30 @@
+package frc.trigon.robot.subsystems.turret;
+
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.turret.kablamaturret.KablamaTurretIO;
+import frc.trigon.robot.subsystems.turret.simulationturret.SimulationTurretIO;
+import org.littletonrobotics.junction.AutoLog;
+
+public class TurretIO {
+    static TurretIO generateIO() {
+        if (RobotConstants.IS_REPLAY)
+            return new TurretIO();
+        if (RobotConstants.ROBOT_TYPE == RobotConstants.RobotType.KABLAMA)
+            return new KablamaTurretIO();
+        return new SimulationTurretIO();
+    }
+
+    protected void updateInputs(TurretInputsAutoLogged inputs) {
+    }
+
+    protected void calculateMotorVoltage(double voltage) {
```

Comment:
The name and the variable are very misleading. Just name this set target angle and make this accept a rotation2d